docs/DeveloperGuide.md

```diff
@@ -1,34 +1,330 @@
+#### [2. Design & Implementation](#design)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[2.1 Architecture Overview](#overview)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[2.2 NotUS](#notus)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[2.3 ParserManager](#parserManager)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[2.4 Commands](#commands)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[2.5 Notebook](#note)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[2.6 Timetable](#event)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[2.7 Tags](#tag)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[2.8 Storage](#storage)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[2.9 User Interface](#ui)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[2.10 System Exception](#exception)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[2.11 Usage of External Libraries](#color)
+#### [3. Product Scope](#scope)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[3.1 Target User Persona](#userpersona)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[3.2 Target User Profile](#userprofile)
+##### &nbsp;&nbsp;&nbsp;&nbsp;&nbsp;&nbsp;[3.3 Value Proposition](#value)
+#### [4. User Stories](#userstories)
+#### [5. Non-Functional Requirements](#nfr)
+#### [6. Glossary](#gloss)
+#### [7. Instructions for Manual Testing](#testinstr)
 
-{Describe the design and implementation of the product. Use UML diagrams and short code snippets where applicable.}
+<br>
 
+## <a id="intro">1. Introduction</a>
 
-## Product scope
-### Target user profile
+NotUS is a quick and simple, Command Line Interface (CLI) based, note-taking application for keyboard-inclined users. NotUS allows for users to categorize notes by tagging as well as pinning the more important notes. NotUS is also designed to assist in planning timetables to highlight possible clashes.
 
-{Describe the target user profile}
+This document describes the design, implementation and architecture of NotUS. The aim of this developer guide is to get developers and potential contributors familiarised with the design and implementation of NotUS. It is assumed that the reader has some basic understanding of UML Notations. If you do not possess such knowledge, this document is probably not meant for you. Please access the [User Guide](https://github.com/AY2021S1-CS2113-T13-1/tp/blob/master/docs/UserGuide.md) instead.
 
-### Value proposition
+#### <a id="setup"><ins>1.1 Setting Up</ins></a>
 
-{Describe the value proposition: what problem does it solve?}
+**Prerequisites:**
+* JDK 11
+* IntelliJ IDE
 
-## User Stories
+Fork this repo and clone it onto your local machine.
+Import the project as a **Gradle project**.
+Ensures that you are using the correct JDK version (For this project we are using JDK 11).
 
-|Version| As a ... | I want to ... | So that I can ...|
+For a more detailed set of instructions, please refer to the [following guide](https://github.com/AY2021S1-CS2113-T13-1/tp/blob/master/README.md).
+
+<br>
+
+## <a id="design">2. Design & Implementation</a>
```

Comment:
You may use ## 2. Design & Implementation<a id="design"></a> for better appearance. The current one will behave like some hyperlink that you cannot click when you move your mouse onto it.
![image](https://user-images.githubusercontent.com/55589340/97667946-f4035e00-1abb-11eb-8b04-db47532a95cd.png)
![image](https://user-images.githubusercontent.com/55589340/97668026-139a8680-1abc-11eb-86b1-0ceda48e84eb.png)